x-pack/plugins/ml/public/application/components/model_snapshots/revert_model_snapshot_flyout/create_calendar.tsx

```diff
@@ -0,0 +1,289 @@
+                        onChange={(d) => setEndDate(d, i)}
+                      />
+                    </EuiFormRow>
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+                <EuiSpacer size="s" />
+                <EuiFlexGroup>
+                  <EuiFlexItem>
+                    <EuiFormRow
+                      fullWidth
+                      label={i18n.translate(
+                        'xpack.ml.revertModelSnapshotFlyout.createCalendar.descriptionLabel',
+                        {
+                          defaultMessage: 'Description',
+                        }
+                      )}
+                    >
+                      <EuiFieldText
+                        style={{ width: '100%', maxWidth: 'inherit' }}
+                        value={c.description}
+                        onChange={(e) => setDescription(e.target.value, i)}
+                      />
+                    </EuiFormRow>
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+              </EuiFlexItem>
+              <EuiFlexItem
+                grow={false}
+                style={{
+                  borderLeft: `1px solid ${euiTheme.euiColorLightShade}`,
+                  marginRight: '0px',
+                }}
+              />
+              <EuiFlexItem grow={false}>
+                <EuiButtonIcon
+                  style={{ margin: 'auto' }}
+                  color={'danger'}
+                  onClick={() => removeCalendarEvent(i)}
+                  iconType="trash"
+                  aria-label={i18n.translate(
+                    'xpack.ml.revertModelSnapshotFlyout.createCalendar.deleteLabel',
+                    {
+                      defaultMessage: 'Delete event',
+                    }
+                  )}
+                />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiPanel>
+          <EuiSpacer size="m" />
+        </div>
+      ))}
+    </>
+  );
+};
+
+interface ChartProps {
+  eventRateData: LineChartPoint[];
+  anomalies: Anomaly[];
+  loading: boolean;
+  onBrushEnd(area: XYBrushArea): void;
+  overlayRanges: Array<{ start: number; end: number }>;
+}
+
+const Chart: FC<ChartProps> = memo(
+  ({ eventRateData, anomalies, loading, onBrushEnd, overlayRanges }) => (
+    <EventRateChart
+      eventRateChartData={eventRateData}
+      anomalyData={anomalies}
+      loading={loading}
+      height={'100px'}
+      width={'100%'}
+      fadeChart={true}
+      overlayRanges={overlayRanges.map((c) => ({
+        start: c.start,
+        end: c.end,
+        color: '#0000ff',
+        showMarker: false,
+      }))}
+      onBrushEnd={onBrushEnd}
+    />
+  ),
+  (prev: ChartProps, next: ChartProps) => {
+    // only redraw if the calendar ranges have changes
+    return (
+      prev.overlayRanges.length === next.overlayRanges.length &&
+      JSON.stringify(prev.overlayRanges) === JSON.stringify(next.overlayRanges)
```

Comment:
`JSON.stringify` seems like not the best solution. Probably a parent component should update the instance of  `overlayRanges ` only when it actually changed.

Reply:
`overlayRanges` is a very simple array of objects and so shouldn't be a problem comparing with `JSON.stringify` 
Will change in a follow up if we see a performance issue.